cpp/src/io/text/multibyte_split.cu

```diff
@@ -418,12 +418,15 @@ std::unique_ptr<cudf::column> multibyte_split(cudf::io::text::data_chunk_source
   auto reader = source.create_reader();
   reader->skip_bytes(relevant_offset_first);
 
-  auto relevant_bytes = reader->get_next_chunk(string_chars_size, stream);
-
-  thrust::copy(rmm::exec_policy(stream),
-               relevant_bytes->data(),  //
-               relevant_bytes->data() + relevant_bytes->size(),
-               string_chars.begin());
+  for (int32_t i = 0; i < string_chars_size; i += ITEMS_PER_CHUNK) {
+    auto read_size = std::min(static_cast<int64_t>(ITEMS_PER_CHUNK), string_chars_size - i);
```

Comment:
Is `string_chars_size` an `int32_t` or an `int64_t`? It seems like all the `auto` variables above are `int32_t` if I were to guess. Just trying to figure out why this cast to `int64_t` is needed.

Reply:
`string_chars_size` is `int64_t` but `ITEMS_PER_CHUNK` is `int32_t`. Do you really get a warning for an implicit upcast to 64-bit? If not, you could drop the `static_cast`.

Reply:
Has std::min been fixed in the more recent C++ standards? Otherwise this is likely a type deduction issue. My go-to solution is usually
```suggestion
    auto read_size = std::min<int64_t>(ITEMS_PER_CHUNK, string_chars_size - i);
```